src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideRequestController.java

```diff
@@ -0,0 +1,132 @@
+package edu.ucsb.cs156.gauchoride.controllers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.ucsb.cs156.gauchoride.entities.RideRequest;
+import edu.ucsb.cs156.gauchoride.repositories.RideRequestRepository;
+
+import edu.ucsb.cs156.gauchoride.errors.EntityNotFoundException;
+
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+
+import javax.validation.Valid;
+
+@Api(description = "Ride request information (admin, driver, rider)")
+@RequestMapping("/api/riderequests")
+@RestController
+public class RideRequestController extends ApiController {
+    @Autowired
+    RideRequestRepository rideRequestRepository;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @ApiOperation(value = "Get a list of all ride requests")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_DRIVER')")
+    @GetMapping("/all")
+    public ResponseEntity<String> rideRequestsAdmin()
+            throws JsonProcessingException {
+        Iterable<RideRequest> rideRequests = rideRequestRepository.findAll();
+        String body = mapper.writeValueAsString(rideRequests);
+        return ResponseEntity.ok().body(body);
+    }
+
+    @ApiOperation(value = "Get a list of all ride requests for current rider")
+    @PreAuthorize("hasRole('ROLE_RIDER')")
+    @GetMapping("/all_rider")
+    public ResponseEntity<String> rideRequestsRider(
+        @ApiParam(name="userid", type="Long", value="userid of the rider", example="123", required=true) @RequestParam Long userid
+        )
```

Comment:
This is not the correct way to implement this.  Instead of passing in the id for the rider, the backend should get the id of the current rider.

There are examples of this in both proj-courses, and proj-happycows; I'll find a few and link to them. 

The way you've implemented this, there is a security hole where any rider with knowledge of how backends work can hack the backend and get any riders rides, which is a privacy violation.   So we need to fix this before its mergeable.  Fortunately, it's an easy fix.